bazel/emscripten_toolchain/wasm_cc_binary.bzl

```diff
@@ -105,6 +191,12 @@ def _wasm_binary_outputs(name, cc_target):
 
     return outputs
 
+_wasm_cc_binary_legacy = rule(
```

Comment:
can you set `name = "wasm_cc_binary"` for both of these? 

that way the target type matches for things that care about the rule type.

Reply:
Done! I didn't know that existed. Thanks!

Two questions if you don't mind answering:
1) Is it not a problem that both `_wasm_cc_binary_legacy` and `_wasm_cc_binary` have the same `name`?
2) Is it recommended that the rule `name` attribute is set for any rule that is wrapped in this style? I'd like to update my internal rules if that's the case. I'm unsure what the consequences of not doing it is.

Reply:
1) This is basically the only use for the `name` parameter AFAIK--when you need to do something spicy like this.
See also: [cc_test.bzl's usage](https://github.com/bazelbuild/bazel/blob/fc5bf5459a52d1ea68b3dcd9fa582de118e452d1/src/main/starlark/builtins_bzl/common/cc/cc_test.bzl#L116-L141)

2) I dunno if it's "recommended" or not, but when you care about the rule type being the same (as, in this case, we do -- for things like `bazel query "kind(wasm_cc_binary...)"` or `native.existing_rules`), you have two options: 
  a) either use `name` or 
  b) do some file import dance like [cc_binary does](https://github.com/bazelbuild/bazel/blob/fc5bf5459a52d1ea68b3dcd9fa582de118e452d1/src/main/starlark/builtins_bzl/common/cc/cc_binary_wrapper.bzl#L21-L37). `cc_binary` does this thing where it defines the rules in separate files so they can both be named `cc_binary`, but then imports them with a different name to do this macro wrap. 

I personally prefer using `name`, but I suppose both are _correct_ so it might just be a matter of style.